cookbook/case_studies/feature_engineering/feast_integration/feast_type_transformers.py

```diff
@@ -0,0 +1,146 @@
+from flytekit.configuration import aws
+from datetime import datetime
+import pandas as pd
+import os
+from typing import Type
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+from feast import repo_config
+from feast.feature_store import FeatureStore
+from feast.repo_config import RepoConfig
+from flytekit import FlyteContext
+from flytekit.core.type_engine import TypeEngine, TypeTransformer
+from flytekit.models.literals import Literal, Scalar
+from flytekit.models.types import LiteralType, SimpleType
+from feast.infra.offline_stores.file import FileOfflineStoreConfig
+from feast.infra.online_stores.sqlite import SqliteOnlineStoreConfig
+from feast import FeatureStore as FeastFeatureStore
+from google.protobuf.struct_pb2 import Struct
+from google.protobuf.json_format import MessageToDict
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.feature_service import FeatureService
+
+
+@dataclass_json
+@dataclass
+class FeatureStoreConfig:
+    registry_path: str
+    project: str
+    s3_bucket: str
+    online_store_path: str = 'online.db'
+
+
+@dataclass_json
+@dataclass
+class FeatureStore:
+    config: FeatureStoreConfig
+
+    def _build_feast_feature_store(self):
+        os.environ["FEAST_S3_ENDPOINT_URL"] = aws.S3_ENDPOINT.get()
+        os.environ["AWS_ACCESS_KEY_ID"] = aws.S3_ACCESS_KEY_ID.get()
+        os.environ["AWS_SECRET_ACCESS_KEY"] = aws.S3_SECRET_ACCESS_KEY.get()
+
+        config = RepoConfig(
+            registry=f"s3://{self.config.s3_bucket}/{self.config.registry_path}",
+            project=self.config.project,
+            # Notice the use of a custom provider.
+            provider="custom_provider.provider.FlyteCustomProvider",
+            offline_store=FileOfflineStoreConfig(),
+            online_store=SqliteOnlineStoreConfig(path=self.config.online_store_path),
```

Comment:
Can we have "redis" online store?

Reply:
Do you mean in addition to the sqlite store? I don't see why not. We could expose that as a configuration.

Reply:
Yeah, we could have it in addition or replace the SQLite one. Both should work!